doc/api/esm.md

```diff
@@ -156,7 +156,7 @@ typically configured server.
 ### URLs
 
 ES modules are resolved and cached as URLs. This means that files containing
-special characters such as `#` and `?` need to be escaped.
```

Comment:
FWIW, the word _escape_ does not appear in the relevant MDN doc, so maybe "encoded" is the way to go instead of "escaped" or "replaced with escape sequences". Also, and I know this wasn't introduced in this PR, but "files containing" makes it sound we're talking about the file contents when we're talking about the file name/path.

```suggestion
ES modules are resolved and cached as URLs. This means that characters such as 
`#` and `? need to be [percent-encoded][].
```

Then add to the bottom:

```
[percent-encoding]: https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding